app/models/host/discovered.rb

```diff
@@ -81,6 +81,8 @@ def populate_fields_from_facts(facts = self.facts_hash, ignored_type)
     unless primary_ip.nil?
       subnet = Subnet.subnet_for(primary_ip)
       if subnet
+        # reload subnet to fetch all references
+        subnet.reload
```

Comment:
This is to prevent N+1, @ares showed me this trick. Marek, do you remember which code in particular is causing the N+1 errors? I vaugly remember this is the interface detection code